packages/cardpay-sdk/sdk/token-bridge-home-side.ts

```diff
@@ -229,11 +232,16 @@ export default class TokenBridgeHomeSide implements ITokenBridgeHomeSide {
         `Timed out waiting for tokens to be bridged to layer 2 for safe owned by ${recipientAddress} after block ${fromBlock}`
       );
     }
+    // TODO: can this use waitForSubgraphIndex?
     let {
       transaction: { id: txnHash },
     } = receivedBridgedTokens[0];
     let receipt = await this.layer2Web3.eth.getTransactionReceipt(txnHash);
-    return receipt;
+    if (receipt.status === true) {
+      return receipt as SuccessfulTransactionReceipt;
+    } else {
+      throw new Error(`Transaction with hash "${txnHash}" was reverted`);
```

Comment:
Is there any other information available in this circumstance that would be useful to include as part of the error?

Reply:
I'm not very clear what revert reasons we should expect for this scenario, do you know any that we need special cases for @habdelra?

Reply:
I don't think these txn's will actually ever revert since they are created programmatically via bridge validators. These types of txn's are use an ethereum `call()` API which allows the client to perform a dry run of the txn before the real txn. If the `call()` reverts, then the bridge validator will never actually send the txn for real (our relay server works the same way).